src/main/java/com/cloudogu/wiki/RefreshWikisServlet.java

```diff
@@ -0,0 +1,36 @@
+package com.cloudogu.wiki;
+
+import com.cloudogu.wiki.scmm.SessionCacheScmWikiListStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * @author Maren Süwer
+ */
+
+public class RefreshWikisServlet extends HttpServlet {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RefreshWikisServlet.class);
+    private final String refreshUrl;
+
+    public RefreshWikisServlet() {
+        refreshUrl = "/smeagol";
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+        request.getSession(true).removeAttribute(SessionCacheScmWikiListStrategy.class.getName());
+        LOG.info("wikis are reloaded");
+        request.getSession().invalidate();
+        try {
+            response.sendRedirect(refreshUrl);
+        } catch (IOException ex) {
+            LOG.warn("Failed to send redirect.", ex);
```

Comment:
What happens, if the IOException is thrown during a redirect? The user gets a white page and nothing happens.
Just do not catch this exception, let it bubble up.